skgstat/MetricSpace.py

```diff
@@ -0,0 +1,120 @@
+import scipy.spatial
+import scipy.spatial.distance
+import scipy.sparse
+import numpy as np
+
+def sparse_dok_get(m, fill_value=np.NaN):
+    mm = np.full(m.shape, fill_value)
+    for (x, y), value in m.items():
+        mm[x,y] = value
+    return mm
+
+class DistanceMethods(object):
+    def find_closest(self, idx, max_dist = None, N = None):
+        """Find the (N) closest points (in the right set) to the point with
+        index idx (in the left set).
```

Comment:
Is max_dist the same as `maxlag`? 
What is the left and right set here referring to?

Reply:
Yes it is, but for the limited usage of distance calculations, i.e. the usage within this class.
Left & right are applicable to MetricSpacePair(), where you have two sets of points, left being transform points and right being observations: https://github.com/mmaelicke/scikit-gstat/blob/metric-space/skgstat/Kriging.py#L295 : `MetricSpacePair(self.transform_coords, self.coords)`

Reply:
So you would argue, that `max_dist` as an argument name makes more sense here, to distinguish it from the `maxlag` parameter?

What happens if I do something like this:

```Python
ms = MetricSpace(coords, max_dist=50)
V = Variogram(ms, values, maxlag=80)
```
Do I miss something, or does the `maxlag=80` not have any effect anymore? Sorry for missing the forest through the trees...

Reply:
Yeah, but it's not super important that they're different.

There is an assert that would flag this situation. If you supply a MetricSpace to Veriogram, you can only do one of these:

* Set MetricSpace.max_dist (this will propagate to maxlag in Variogram)
* Set Variogram.maxlag (this will be honored, but won't optimize distance calculations)
* Set both to the same value

Reply:
@redhog  what about adding a warning if maxlag and max_dist are both set, but not of same value? 
A warning will break the code, but the user is able to ignore it...

Reply:
Sounds good to me.

Reply:
Possibly a different warning if that's the case and maxlag > max_dist?